src/compiler/debug.ts

```diff
@@ -1,414 +1,341 @@
+                    return enumName;
                 }
             }
-
-            return stableSort<[number, string]>(result, (x, y) => compareValues(x[0], y[0]));
         }
-
-        export function formatSyntaxKind(kind: SyntaxKind | undefined): string {
-            return formatEnum(kind, (<any>ts).SyntaxKind, /*isFlags*/ false);
-        }
-
-        export function formatNodeFlags(flags: NodeFlags | undefined): string {
-            return formatEnum(flags, (<any>ts).NodeFlags, /*isFlags*/ true);
-        }
-
-        export function formatModifierFlags(flags: ModifierFlags | undefined): string {
-            return formatEnum(flags, (<any>ts).ModifierFlags, /*isFlags*/ true);
-        }
-
-        export function formatTransformFlags(flags: TransformFlags | undefined): string {
-            return formatEnum(flags, (<any>ts).TransformFlags, /*isFlags*/ true);
-        }
-
-        export function formatEmitFlags(flags: EmitFlags | undefined): string {
-            return formatEnum(flags, (<any>ts).EmitFlags, /*isFlags*/ true);
-        }
-
-        export function formatSymbolFlags(flags: SymbolFlags | undefined): string {
-            return formatEnum(flags, (<any>ts).SymbolFlags, /*isFlags*/ true);
-        }
-
-        export function formatTypeFlags(flags: TypeFlags | undefined): string {
-            return formatEnum(flags, (<any>ts).TypeFlags, /*isFlags*/ true);
-        }
-
-        export function formatObjectFlags(flags: ObjectFlags | undefined): string {
-            return formatEnum(flags, (<any>ts).ObjectFlags, /*isFlags*/ true);
-        }
-
-        let isDebugInfoEnabled = false;
-
-        interface ExtendedDebugModule {
-            init(_ts: typeof ts): void;
-            formatControlFlowGraph(flowNode: FlowNode): string;
-        }
-
-        let extendedDebugModule: ExtendedDebugModule | undefined;
-
-        function extendedDebug() {
-            enableDebugInfo();
-            if (!extendedDebugModule) {
-                throw new Error("Debugging helpers could not be loaded.");
+        return value.toString();
+    }
+    function getEnumMembers(enumObject: any) {
+        const result: [number, string][] = [];
+        for (const name in enumObject) {
+            const value = enumObject[name];
+            if (typeof value === "number") {
+                result.push([value, name]);
             }
-            return extendedDebugModule;
-        }
-
-        export function printControlFlowGraph(flowNode: FlowNode) {
-            return console.log(formatControlFlowGraph(flowNode));
         }
-
-        export function formatControlFlowGraph(flowNode: FlowNode) {
-            return extendedDebug().formatControlFlowGraph(flowNode);
+        return stableSort<[number, string]>(result, (x, y) => compareValues(x[0], y[0]));
+    }
+    export function formatSyntaxKind(kind: SyntaxKind | undefined): string {
+        return formatEnum(kind, (<any>ts).SyntaxKind, /*isFlags*/ false);

```

Comment:
Hi.
Not pretty sure, but I think (<any>ts).SyntaxKind could not get the enum Object. (In my simple test case, it is undefined)
I use a new script to generate a new file like diagnosticMessages.generated.json to extract const keyword and get these enum types.
Hope this help!

Reply:
We compile with `preserveConstEnums` to keep them in the resulting file.